Sources/SwiftProtobuf/DoubleParser.swift

```diff
@@ -18,9 +18,14 @@ import Foundation
 internal class DoubleParser {
     // Temporary buffer so we can null-terminate the UTF-8 string
     // before calling the C standard library to parse it.
+    //
     // In theory, JSON writers should be able to represent any IEEE Double
     // in at most 25 bytes, but many writers will emit more digits than
     // necessary, so we size this generously.
+    //
+    // This is also used for TextFormat, there the spec says decimal numbers
+    // can actually go over the limit can get mapped to [-]inifinity; so in
+    // theory this needs to be larger, but we haven't worried about it yet.
```

Comment:
This comment conflates two different issues:
* Overflow and underflow handling.  Overflow (e.g., "1e9999") and underflow (e.g., "1e-9999") handling are pretty well specified by IEEE754 and related standards, so there's not really much need to talk about it here because there's really not much variation in how people handle it.
* The real concern here is how many characters we should parse, which is limited by the size of this buffer.  The comment above about JSON parsing applies equally to TextFormat.

Reply:
I was actually trying (poorly) to capture both points are a problem.

TextFormat specifically calls out handling for overflows for float/double fields:
https://protobuf.dev/reference/protobuf/textformat-spec/#value

> Overflows are treated as infinity or -infinity.

But the JSON protobuf spec doesn't mention anything: https://protobuf.dev/programming-guides/proto3/#json

So the current finite check isn't completely correct, when called for TextFormat.

The other, is since things can be really long and thus overflow, the buffer size could also be a problem.

Reply:
Split/updated the comment to hopefully better capture this.

Reply:
Much better.   Keep these two examples in mind:
* Numeric overflow can happen even with short tokens:  "1e9999" is too large a value for Double and thus will parse to Infinity (in the default rounding mode) even though it's only 6 characters long.
* Overly long tokens aren't necessarily numeric overflows: "123.999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999" is a very long token but does not overflow when correctly parsed as a floating-point number.

The code here has an arbitrary limit of 128 characters for a floating-point value.  I personally think it entirely acceptable to consider any numeric token with more characters to be malformed, as there's never a good reason to have a number token this long.  (This is the sort of corner case that spec writers never think about.  :shrug:)

Reply:
Ok to resolve this, or did you want me to pull in any addition into into the code comment?